crypto/cms/cms_ess.c

```diff
@@ -335,3 +337,165 @@ ASN1_OCTET_STRING *cms_encode_Receipt(CMS_SignerInfo *si)
     CMS_ReceiptRequest_free(rr);
     return os;
 }
+
+/*
+ * Add signer certificate's V2 digest to a SignerInfo
+ * structure
+ */
+
+CMS_SignerInfo *CMS_add1_signing_cert_v2(CMS_SignerInfo *si, X509 *signer,
```

Comment:
This method (and so its next) is over-complicated.
It should have been split in two : 
- Create the Signing Certificate 
- Add it to the CMS Signer Info.
- returning the same pointer  as input is not in the project usages.
It has some Memory management issue, like double-free.